### PR TITLE
Kraken upgrade and base image removal

### PR DESCRIPTION
**Pull request by hunter**
This is a start on the Kraken images and combining base and daemon into a single image.

Untested other than checking that the 14.04 Dockerfile would build. Might need some more cleanup (and likely updates of symlinks). Thought I'd get the changes out there for discussion.

Do we also want to keep the demo image?

### CODE REVIEW

**Comment by leseb** [comment]
I was thinking or at least questioning myself if we shouldn't merge demo inside daemon as well. Thoughts?

File: ceph-releases/kraken/centos/7/demo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# DOCKER-VERSION 1.0.0
+#
+# Ceph Demo AIO
+#
+# VERSION 0.0.1
+
+FROM ceph/base:tag-build-master-kraken-centos-7
```
Comment (leseb) [inline ceph-releases/kraken/centos/7/demo/Dockerfile]: This image won't exist anymore.

File: ceph-releases/kraken/fedora/24/daemon/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# DOCKER-VERSION 1.0.0
```
Comment (leseb) [inline ceph-releases/kraken/fedora/24/daemon/Dockerfile]: Are we sure that f24 has the latest version of Ceph?

File: ceph-releases/kraken/fedora/24/demo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# DOCKER-VERSION 1.0.0
+#
+# Ceph Demo AIO
+#
+# VERSION 0.0.1
+
+FROM ceph/base:tag-build-master-kraken-fedora-24
```
Comment (leseb) [inline ceph-releases/kraken/fedora/24/demo/Dockerfile]: ditto

File: ceph-releases/kraken/redhat/7.2/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
+FROM  registry.access.redhat.com/rhel7.2
```
Comment (leseb) [inline ceph-releases/kraken/redhat/7.2/Dockerfile]: This one is purely Red Hat let's remove it as some of the links won't work, I'll add the support for Red Hat :).

File: ceph-releases/kraken/ubuntu/14.04/daemon/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# DOCKER-VERSION 1.0.0
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/14.04/daemon/Dockerfile]: I'll like to drop the support of 14.04 and make 16.04 the default for Kraken. Can you remove it and stick with 16.04 as a reference for all the images?

**Comment by leseb** [comment]
And yes this is obviously breaking the CI...

File: ceph-releases/kraken/redhat/7.2/Dockerfile (hunk 4 above)
Comment (hunter) [inline ceph-releases/kraken/redhat/7.2/Dockerfile]: Do we want to delete the `redhat` dir for now?

File: ceph-releases/kraken/ubuntu/14.04/daemon/Dockerfile (hunk 5 above)
Comment (hunter) [inline ceph-releases/kraken/ubuntu/14.04/daemon/Dockerfile]: Removed 14.04 and updated the symlinks

File: ceph-releases/kraken/fedora/24/daemon/Dockerfile (hunk 2 above)
Comment (hunter) [inline ceph-releases/kraken/fedora/24/daemon/Dockerfile]: Seems as though Ceph is still at Jewel for Fedora. Do we want to remove for now?

Comment (leseb) [inline ceph-releases/kraken/fedora/24/daemon/Dockerfile]: Yes please.

File: ceph-releases/kraken/redhat/7.2/Dockerfile (hunk 4 above)
Comment (leseb) [inline ceph-releases/kraken/redhat/7.2/Dockerfile]: Yes please.

Comment (hunter) [inline ceph-releases/kraken/redhat/7.2/Dockerfile]: Done :)

File: ceph-releases/kraken/fedora/24/daemon/Dockerfile (hunk 2 above)
Comment (hunter) [inline ceph-releases/kraken/fedora/24/daemon/Dockerfile]: Done :)

File: ceph-releases/kraken/centos/7/daemon/2.sh [hunk 6]
```diff
@@ -0,0 +1 @@
+../../../ubuntu/14.04/daemon/2.sh
```
Comment (leseb) [inline ceph-releases/kraken/centos/7/daemon/2.sh]: Can you point to 16.04 instead? And drop 14.04? Thanks!

Comment (hunter) [inline ceph-releases/kraken/centos/7/daemon/2.sh]: Sorry, had overlooked the Centos symlinks. Updated.

File: ceph-releases/kraken/centos/7/demo/README.md [hunk 7]
```diff
@@ -0,0 +1 @@
+../../../ubuntu/14.04/demo/README.md
```
Comment (leseb) [inline ceph-releases/kraken/centos/7/demo/README.md]: The link points to a non-existing file. Just copy the jewel file for the demo container.

File: ceph-releases/kraken/centos/7/demo/entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1 @@
+../../../ubuntu/14.04/demo/entrypoint.sh
```
Comment (leseb) [inline ceph-releases/kraken/centos/7/demo/entrypoint.sh]: ditto

File: ceph-releases/kraken/ubuntu/16.04/demo/README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+../../../ubuntu/14.04/demo/README.md
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/demo/README.md]: ditto

File: ceph-releases/kraken/ubuntu/16.04/demo/entrypoint.sh [hunk 10]
```diff
@@ -0,0 +1 @@
+../../../ubuntu/14.04/demo/entrypoint.sh
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/demo/entrypoint.sh]: ditto

File: ceph-releases/kraken/centos/7/demo/README.md (hunk 7 above)
Comment (hunter) [inline ceph-releases/kraken/centos/7/demo/README.md]: Darn it... I thought I'd fixed this for Centos. Updated!

**Comment by leseb** [comment]
Thanks @hunter

**Pull request merged**